src/LuaRenderer/RenderState.ts

```diff
@@ -71,6 +73,36 @@ export class RenderState {
 		return this.listNodesStack.pop();
 	}
 
+	/**
+	 * Adds a newline to the end of the string.
+	 * @param text The text.
+	 */
+	public newline(text: string) {
+		return text + "\n";
+	}
+
+	/**
+	 * Prefixes the text with the current indent.
+	 * @param text The text.
+	 */
+	public indented(text: string) {
+		return this.indent + text;
+	}
+
+	/**
+	 * Renders a line, adding the current indent, a semicolon if necessary, and "\n".
+	 * @param text The content of the line.
+	 * @param endNode Node used to determine if a semicolon should be added. Undefined means no semi will be added.
+	 */
+	public line(text: string, endNode?: lua.Statement) {
```

Comment:
I like that `endNode` is optional here! Since only certain nodes required `getEnding`, we don't do extra works on nodes that don't require that check.